### PR TITLE
Fix hoomd.Operations.__len__

### LINKED ISSUE #1128: Maximum recursion depth when taking the length of an Operations instance
## Script

```python
# Include a minimal script that reproduces the problem
import hoomd
operations = hoomd.Operations()
len(operations)
```

## Output

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/joaander/build/hoomd/hoomd/operations.py", line 238, in __len__
    return len(list(self))
  File "/Users/joaander/build/hoomd/hoomd/operations.py", line 238, in __len__
    return len(list(self))
  File "/Users/joaander/build/hoomd/hoomd/operations.py", line 238, in __len__
    return len(list(self))
  [Previous line repeated 995 more times]
RecursionError: maximum recursion depth exceeded
```

## Expected output

No output.

## Configuration

Platform:
- Mac

Installation method:
- Compiled from source

Versions:
- Python version: 3.9.6
- **HOOMD-blue** version: AttributeSetError - likely present on master as well

## Developer

@b-butler can you fix this?

### PR DESCRIPTION
**Pull request by b-butler**
## Description

Fixes the behavior of `hoomd.Operations.__len__`.

## Motivation and context

Resolves #1128

## How has this been tested?

Tests have been added for `__len__` and `__iter__`.

## Change log

```
Fixed: Calling `hoomd.Operations.__len__` no longer raises a `RecursionError`.
```

## Checklist:

- [ ] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/master/CONTRIBUTING.md).
- [ ] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/master/ContributorAgreement.md).
- [ ] My name is on the list of contributors (`sphinx-doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

File: hoomd/operations.py [hunk 1]
```diff
@@ -230,12 +230,12 @@ def __contains__(self, operation):
     def __iter__(self):
         """Iterates through all contained operations."""
         integrator = (self._integrator,) if self._integrator else []
-        yield from chain(self._tuners, self._updaters, integrator,
-                         self._writers, self._computes)
```
Comment (joaander) [inline hoomd/operations.py:234]: Is this change important to fixing the issue? I liked the previous ordering as it follows the same order the operations are applied during `run` (except computes which may be triggered at an time during the step)

File: hoomd/operations.py [hunk 2]
```diff
@@ -230,12 +230,12 @@ def __contains__(self, operation):
...
+        yield from chain(integrator, self._updaters, self._writers,
+                         self._tuners, self._computes)
 
     def __len__(self):
         """Return the number of operations contained in this collection."""
-        return len(list(self))
+        return len([None for _ in self])
```
Comment (joaander) [inline hoomd/operations.py:238]: Why not add the lengths of the contained lists (and add 1 if there is an integrator)?

**Review by joaander — COMMENTED** [review]
Thanks for the fix. I have some questions.

**Review by tcmoore3 — COMMENTED** [review]
Looks good after you address Josh's questions.

File: hoomd/operations.py (hunk 2 above)
Comment (b-butler) [reply hoomd/operations.py:238]: I didn't do this to keep the code somewhat simpler, (utilizing the existing `__iter__`), but there would be nothing wrong with adding list lengths.

File: hoomd/operations.py (hunk 1 above)
Comment (b-butler) [reply hoomd/operations.py:234]: No it isn't, and that is good reason to revert.

**Review by joaander — APPROVED** [review]
Thanks!

**Pull request merged**